### ISSUE TITLE
Incorrect output when using safeBothPrefix with duplicate declarations

### ISSUE THREAD

**Issue by yepitschunked** [elchininet/postcss-rtlcss#185]
Followup to https://github.com/elchininet/postcss-rtlcss/issues/181

The original issue seems only partially fixed in https://github.com/elchininet/postcss-rtlcss/pull/182. Duplicates are moved into `ltr`/`rtl` selectors properly, but the behavior with`[dir]` is broken. Here's a playground example:
https://elchininet.github.io/postcss-rtlcss/#640141a85d223

All of the styles in `.bar` should be moved into `[dir] .bar`

**Comment by elchininet** [comment]
Hi @yepitschunked,
Thanks for reporting. I‘ll take a look at it.
I‘ll let you know when it is solved.
Regards.

**Comment by elchininet** [comment]
Hi @yepitschunked,

This was the output of the input that you used before the latest changes:

```css
[dir] .foo {
    padding-top: 4px;
    padding-bottom: 4px;
    padding-left: 8px;
    padding-right: 8px;
}

/* this is just .foo repeated twice */
.bar {
    padding-top: 4px;
    padding-bottom: 4px;
    padding-left: 8px;
    padding-right: 8px;
}

[dir] .bar {
    padding-top: 4px;
    padding-bottom: 4px;
    padding-left: 8px;
    padding-right: 8px;
}
```

Do you think that this is enough or do you expect that all the declarations are moved even if they are overridden?

**Comment by yepitschunked** [comment]
Yeah, I think this output would be fine. 

In the old output, the `.bar` styles are completely overridden by the `[dir] .bar` styles, so they have the correct precedence when coexisting with `[dir=ltr]` rules. It's probably unnecessary to have the unprefixed `.bar` styles, but it doesn't hurt. 

In the new output, there are styles in `.bar` which aren't replicated in `[dir] .bar`, which means that they will have less precedence than `[dir=ltr]` rules, which breaks the `safeBothPrefix` behavior.

**Comment by elchininet** [comment]
Hi @yepitschunked,

Thanks for the explanation. As I thought there is no a strong requirement to move all the declarations if they have been overridden already.

I‘ll release a patch soon restoring the previous behaviour but keeping what was fixed in the last version.

I‘ll let you know.

Regards.

**Comment by elchininet** [comment]
@yepitschunked,
The issue has been solved, you can download the version [4.0.3 of the package](https://www.npmjs.com/package/postcss-rtlcss/v/4.0.3) which contains the fix.
Thanks again for reporting and for using the library.
Regards!
